language/documentation/book/src/friends.md

```diff
@@ -0,0 +1,126 @@
+# Friends
+
+The `friend` syntax is used to declare modules that are trusted by the current module.
+A trusted module is allowed to call any function defined in the current module that have the `public(friend)` visibility.
+For details on function visibilities, please refer to the *Visibility* section in [Functions](./functions.md).
+
+## Friend declaration
+
+A module can declare other modules as friends via friend declaration statements, in the format of
+
+- `friend <address::name>` — friend declaration using fully qualified module name like the example below, or
+```move
+address 0x42 {
+module A {
+    friend 0x42::B;
+}
+}
+```
+- `friend <module-name-alias>` — friend declaration using a module name alias, where the module alias is introduced via the `use` statement.
+```move
+address 0x42 {
+module A {
+    use 0x42::B;
+    friend B;
+}
+}
+```
+
+A module may have multiple friend declarations, and the union of all the friend modules forms the friend list.
+In the example below, both `0x42::B` and `0x42::C` are considered as friends of `0x42::A`.
+```move
+address 0x42 {
+module A {
+    friend 0x42::B;
+    friend 0x42::C;
+}
+}
+```
+
+Unlike `use` statements, `friend` can only be declared in the module scope and not in the expression block scope.
+In theory, it is valid to have `friend` declarations at any location where a top-level construct (e.g., `use`, `function`, `struct`, etc) is allowed.
+However, for readability, it is advised to place friend declarations near the beginning of the module definition.
+
+Note that the concept of friendship does not apply to Move scripts:
+- A Move script cannot declare `friend` modules as doing so is considered meaningless: there is no mechanism to call the function defined in a script.
+- A Move module cannot declare `friend` scripts as well because scripts are ephemeral code snippets that are never published to global storage.
+
+### Friend declaration rules
+Friend declarations are subject to the following rules:
+
+- A module cannot declare itself as a friend.
+
+```move=
+address 0x42 {
+module M { friend Self; // ERROR! }
+//                ^^^^ Cannot declare the module itself as a friend
+}
+
+address 0x43 {
+module M { friend 0x43::M; // ERROR! }
+//                ^^^^^^^ Cannot declare the module itself as a friend
+}
+```
+
+- Friend modules must be known by the compiler
+```move=
```

Comment:
Some Markdown implementations are more picky about having blank lines between paragraphs, so if it works to have a blank line between the bullets and the example code (as in the first bullet above), I would recommend to do that consistently. Likewise, for a blank line after the "Friend declaration rules" section header above.

Reply:
Will add a blank line in both cases.